lib/ops/flatten.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {Attribute} from '../attribute';
+import {InferenceHandler} from '../backend';
+import {Operator} from '../operators';
+import {Tensor} from '../tensor';
+
+export abstract class Flatten implements Operator {
+  abstract run(inferenceHandler: InferenceHandler, inputs: Tensor[]): Tensor[]|Promise<Tensor[]>;
+
+  initialize(attributes: Attribute): void {
+    this.axis = attributes.getInt('axis', 1);  // default axis is 1
+  }
+
+  checkInputs(inputs: Tensor[]): boolean {
+    if (!inputs || inputs.length !== 1) {
+      return false;
+    }
+
+    if (this.axis < 0 || this.axis >= inputs[0].dims.length) {
+      return false;
+    }
+
```

Comment:
Probably should return `false` for a scalar tensor (i.e.) if the input tensor has dims.length == 0.

Reply:
Added a checking statement for scalar tensor.